### ISSUE TITLE
BUG: Segmentation fault when calling pytorch function after np.exp (numpy 1.21.2)

### ISSUE THREAD

**Issue by kokamido** [pytorch/pytorch#79410]
### 🐛 Describe the bug

Hi! There is an issue connected to numpy and pytorch. It reproduces with numpy==1.21.2 from conda.
If I run provided code example with SIZE=15 then both print functions (they are exactly the same) will print True. If I run it with SIZE=20, the first print will display True but the second will crash because of segmentation fault. If I run it with SIZE=1000 it will display True and False. If I remove np.exp call the code will print True True for any positive int SIZE.

```
import torch
import numpy as np

SIZE=19

print(torch.all(torch.isfinite(torch.fft.fft(torch.eye(SIZE), dim=1))))
np.exp([2])
print(torch.all(torch.isfinite(torch.fft.fft(torch.eye(SIZE), dim=1))))
```

This behavior can be reproduced in the following docker container:

```
FROM ubuntu:focal-20220531

RUN apt update

RUN apt install -y wget

# Miniconda installation
RUN wget --quiet https://repo.anaconda.com/miniconda/Miniconda3-py39_4.11.0-Linux-x86_64.sh -O ~/miniconda.sh && \
    /bin/bash ~/miniconda.sh -b -p /opt/conda && \
    rm ~/miniconda.sh

RUN /opt/conda/bin/conda install  -c pytorch -c nvidia pytorch==1.10.1

RUN /opt/conda/bin/conda install  -c pytorch -c nvidia numpy==1.21.2

ENTRYPOINT /bin/bash
```

There are a few related things in the [numpy issue](https://github.com/numpy/numpy/issues/20356), especially in [this comment](https://github.com/numpy/numpy/issues/20356#issuecomment-970716796). I reported a [similar issue](https://github.com/numpy/numpy/issues/21714) to numpy but
it looks like some pytorch functions are affected by the CPU registers state so I think the current issue may be useful here too.

### Versions

numpy==1.21.2
pytorch==1.10.1

cc @mruberry @rgommers

**Comment by vadimkantorov** [comment]
As further found by @kokamido, opencv works around gcc/avx issues by doing register clean up before/after ops (and maybe detect/report incorrect state): https://github.com/opencv/opencv/pull/12964

maybe PyTorch should do the same to guard against gcc bugs and similar bugs in user code dependencies

**Comment by albanD** [comment]
Note that, from the linked numpy issue, this should be fixed with numpy >=1.22 as there was indeed an issue with their exp op.
Tentatively closing this. Feel free to re-open if you see this issue with a recent numpy version!

**Comment by vadimkantorov** [comment]
I wonder if there're similar issues with AVX512 / mask registers in PyTorch

**Comment by anywallsocket** [comment]
Importing numpy after torch has solved this issue for me
Or, as has been said, upgrading from numpy 1.21.5 to 1.22.0

**Comment by vadimkantorov** [comment]
I would say, PyTorch 2.0 should not allow being used with numpy <  1.22.0, or at least show a warning...
